src/wasm/wasm.cpp

```diff
@@ -1361,4 +1361,90 @@ void Module::updateMaps() {
 
 void Module::clearDebugInfo() { debugInfoFileNames.clear(); }
 
+// UniqueNameMapper
```

Comment:
IMO it would make sense to introduce a new .cpp file to correspond to parsing.h. If that were a separate PR, it would make sense to move the rest of the implementation code in parsing.h into the new .cpp file as well.

Reply:
Fair enough, that is worth the work. Done in #3635.